### PR TITLE
772 frontend create component to render all events

### PR DESCRIPTION
**Pull request by bcho892**
# Event Page (User View)

Created most logic for displaying the events, test by checking for any bad behaviour on the `/events` route.

## Detailed View
image

## Sign up area

image

## Preview events for upcoming and current

>[!NOTE]
>Upcoming events are sorted ascending (so closest to "now"), a past event is defined as the end date elapsing or a day passing from the start date if no end date is given

image

## Past Events

image

### CODE REVIEW

File: client/src/components/generic/Event/EventDetailed/EventDetailed.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+      <div className="flex w-full flex-col items-center justify-center gap-4">
+        <div className="flex h-fit w-full justify-center border bg-white">
+          <Image
+            src={image}
+            alt="Event cover image"
+            width={500}
+            height={300}
+            className="block h-auto max-h-[400px] w-auto  py-4"
+          />
+        </div>
```
Comment (jeffplays2005) [inline client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: image

Is it intentional that the border doesn't wrap the image and overlaps in a rectangle? Thought just as an improvement, could just have the image centre instead.

File: client/src/components/generic/Event/EventUtils.ts [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+   */
+  signUpOpen: (signUpOpenDate: Date) => {
+    const currentDate = new Date()
+    // Sign Ups have opened
+    if (signUpOpenDate <= currentDate) {
+      return `Sign ups have opened! (${EventDateFormatting.getRelativeTimeString(signUpOpenDate)})` as const
+    } else {
+      return `Sign ups open ${EventDateFormatting.getRelativeTimeString(signUpOpenDate)} (${EventDateFormatting.shortDate(signUpOpenDate)})` as const
+    }
+  },
```
Comment (jeffplays2005) [inline client/src/components/generic/Event/EventUtils.ts:24]: Super nice to have this although think we can have a date range or something that specifies the ending date of signups?

File: client/src/components/generic/Event/EventDetailed/EventDetailed.tsx (hunk 1 above)
Comment (jeffplays2005) [inline client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: Just out of curiosity as well, we might need Miguel for some designing here as the image being detached from the main div makes it a bit difficult to navigate, could we instead have the image on the side?

**Review by jeffplays2005 — COMMENTED** [review]
Just a bunch of nit picks on some designing here and there but everything looks quite solid! 

Also wondering, could we have the events instead being on the same page as /events, but use /events/{id} instead? Just thought it'd make sharing easier

File: client/src/components/generic/Event/EventUtils.ts (hunk 2 above)
Comment (bcho892) [reply client/src/components/generic/Event/EventUtils.ts:24]: the client never close sign ups anyway, so for us we just close it after the event is over

File: client/src/components/generic/Event/EventDetailed/EventDetailed.tsx (hunk 1 above)
Comment (bcho892) [reply client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: That was part of the figma design, I'd argue that on the side would squish the text too much, however it being on the bottom may work 

image

Comment (bcho892) [reply client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: the image should ideally fill the container so yes its not intentional, will need to be resolved

**Comment by bcho892** [comment]
> Also wondering, could we have the events instead being on the same page as /events, but use /events/{id} instead? Just thought it'd make sharing easier

Can probably use a query param (i.e `/events?id={id}`), as that's the only way with static export

File: client/src/components/generic/Event/EventDetailed/EventDetailed.tsx (hunk 1 above)
Comment (jeffplays2005) [reply client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: Sounds good

File: client/src/components/generic/Event/EventUtils.ts (hunk 2 above)
Comment (bcho892) [reply client/src/components/generic/Event/EventUtils.ts:24]: but yea if they decide to put it in may be worth

**Review by asun555 — APPROVED** [review]
LGTM

**Review by zlrkw11 — APPROVED** [review]
clean

File: client/src/components/generic/Event/EventDetailed/EventDetailed.tsx (hunk 1 above)
Comment (bcho892) [reply client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: done

Comment (bcho892) [reply client/src/components/generic/Event/EventDetailed/EventDetailed.tsx:94]: done

**Review by jeffplays2005 — COMMENTED** [review]
One last pick as this is highly unlikely but still possible, is there any default asset that we can use in place if no event cover image is provided?
image
image

Thought it would make the website look more "professional"

**Comment by bcho892** [comment]
> One last pick as this is highly unlikely but still possible, is there any default asset that we can use in place if no event cover 
> Thought it would make the website look more "professional"

https://placehold.co/600x400?text=UASC+Event

**Comment by jeffplays2005** [comment]
> > One last pick as this is highly unlikely but still possible, is there any default asset that we can use in place if no event cover
> > Thought it would make the website look more "professional"
> 
> https://placehold.co/600x400?text=UASC+Event

I tried running this locally although the client doesn't seem to allow for svgs/xml coming from external sources. I believe this is probably 
image

image

**Comment by bcho892** [comment]
You should check the preview link which has the latest changes

**Review by jeffplays2005 — APPROVED** [review]

**Pull request merged**